tests/main/interfaces-iscsi-initiator-support/task.yaml

```diff
@@ -0,0 +1,110 @@
+summary: Ensure that the iscsi-initiator-support interface works.
+
+details: |
+  The iscsi-initiator-support interface allows snaps to act as iSCSI initiators,
+  enabling them to discover, connect to, and manage iSCSI targets for
+  block storage access.
+
+  A snap which defines an iscsi-initiator-support plug must be shown in the interfaces
+  list. The plug must not be auto-connected on install and, as usual, must be
+  able to be reconnected.
+
+  A snap declaring a plug on this interface must be able to access iSCSI
+  configuration files, ConfigFS for Linux-IO (LIO) target management,
+  and communicate with the iSCSI daemon.
+
+# Disable tests for core environments
+systems: [-ubuntu-core-*]
+
+environment:
+  TEST_SNAP: test-iscsi-initiator-support-interfaces
+
+prepare: |
+  echo "Create test iSCSI configuration directory structure"
+  "$TESTSTOOLS"/fs-state mock-dir /etc/iscsi/nodes
+  tests.cleanup defer "$TESTSTOOLS"/fs-state restore-dir /etc/iscsi/nodes
+  "$TESTSTOOLS"/fs-state mock-file /etc/iscsi/initiatorname.iscsi
+  tests.cleanup defer "$TESTSTOOLS"/fs-state restore-file /etc/iscsi/initiatorname.iscsi
+  "$TESTSTOOLS"/fs-state mock-file /etc/iscsi/iscsid.conf
+  tests.cleanup defer "$TESTSTOOLS"/fs-state restore-file /etc/iscsi/iscsid.conf
+  echo "InitiatorName=iqn.1993-08.org.debian:01:abcdef123" > /etc/iscsi/initiatorname.iscsi
+  echo "node.startup = automatic" > /etc/iscsi/iscsid.conf
+
+  snap install snapcraft --channel="${SNAPCRAFT_SNAP_CHANNEL}" --classic
+  tests.cleanup defer snap remove --purge snapcraft
```

Comment:
snaps installed on tests are automatically removed, no need to include that in the test